src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -149,6 +145,23 @@ std::shared_ptr<Table> SyntheticTableGenerator::generate_table(
             }
           }
 
+          std::vector<bool> null_values;
+
+          /**
+           * If a ratio of to-be-created NULL values is given, fill the null_values vector used in the ValueSegment
+           * constructor in a regular interval based on the null_ratio with true.
+           */
+          if (column_specifications[column_index].null_ratio) {
+            null_values = std::vector<bool>(chunk_size, false);
+
+            const double step_size = 1.0 / column_specifications[column_index].null_ratio.value();
```

Comment:
We should probably use step_size as int here. Check that this still gives the right ratio in the end.